MetaMask/metamask-extension
# UX: Snaps: Only display Notification in Global Menu if enabled snap uses that permission

## Explanation

The Snaps™ team has asked that we only display the "Notifications" global menu item *if* the user has a snap™ that uses that permission.

Reference:  https://github.com/MetaMask/metamask-extension/issues/20911

## Screenshots/Screencaps

### With a Snap™ enabled that uses `snap_notify` permission


<img width="267" alt="SCR-20230915-kmso" src="https://github.com/MetaMask/metamask-extension/assets/46655/f5b5c70e-e7fd-4597-b149-dc1ccb6c931b">

### With no Snap™ enabled that uses `snap_notify` permission

<img width="281" alt="SCR-20230915-kmqi" src="https://github.com/MetaMask/metamask-extension/assets/46655/b1ac1a2b-b40a-4ac1-8812-82ef5e1c56bd">


## Manual Testing Steps

1.  Open the Global Menu, don't see "Notifications"
2. Go to https://snaps.metamask.io/ , install the "Push V1" snap™ and/or the WalletChat snap™
3. See the "Notifications" item in the Global Menu
4. Go to Settings > Snaps, disable both snaps™
5. Open the Global menu, don't see "Notifications"

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
